fem/restriction.cpp

```diff
@@ -695,8 +695,8 @@ H1FaceRestriction::H1FaceRestriction(const FiniteElementSpace &fes,
 #endif
 
    // If fespace == H1
-   const FiniteElement *fe = fes.GetFE(0);
-   const TensorBasisElement *tfe = dynamic_cast<const TensorBasisElement*>(fe);
+   const FiniteElement *fe0 = fes.GetFE(0);
```

Comment:
There are many different `fe` names in this function, maybe @YohannDudouit should look at it too.

Reply:
This code has been completely refactored in #2195, I think it would be wiser to resolve these shadowing after the merge, no?